### PR TITLE
Add check for kubelet is running in the initial use namespace

### PR DESCRIPTION
**Pull request by chansuke**
#### What type of PR is this?

/kind feature

##### What this PR does
The new function `checkInitialUserNamespace()` checks if the `Kubelet` process is running in the initial user namespace.
The check is performed by reading the /proc/self/uid_map file, which describes the UID map for the current process, and comparing its contents to the expected UID map of the initial user namespace, which is "0 0 4294967295".

##### Why we need it
The reason why I think it's important for the `Kubelet` to run in the initial user namespace is because it ensures that the `Kubelet` has the same view of users and groups as the host system.
This is vital for maintaining the security and integrity of the system, as it prevents processes running within the `Kubelet` from potentially gaining unauthorized access to system resources by manipulating their perceived user or group identities.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubelet/app/server_others.go [hunk 1]
```diff
@@ -28,7 +28,25 @@ func checkPermissions() error {
...
+
+// checkInitialUserNamespace checks if kubelet is running in the initial user namespace.
+// http://man7.org/linux/man-pages/man7/user_namespaces.7.html
+func checkInitialUserNamespace() error {
+	uidMap, err := ioutil.ReadFile("/proc/self/uid_map")
+	if err != nil {
+		return err
+	}
+
+	if strings.TrimSpace(string(uidMap)) != "0\t0\t4294967295" {
```
Comment (SataQiu) [inline cmd/kubelet/app/server_others.go:47]: How about adding some comments about the expected content of `/proc/self/uid_map`?
For example:
```
// When running in the initial host's user namespace, the `/proc/self/uid_map` file content looks like:
// 0          0 4294967295
```

**Review by SataQiu — CHANGES_REQUESTED** [review]

**Comment by dims** [comment]
Please add detailed comments on why this is needed and what it does? beyond the title there isn't any context :(

File: cmd/kubelet/app/server_others.go (hunk 1 above)
Comment (chansuke) [reply cmd/kubelet/app/server_others.go:47]: @SataQiu Thank you so much for your review, I will add some comment.

**Comment by chansuke** [comment]
@dims Thank you for your comment and sorry for the inconvenience. I have updated the description(`What this PR does / why we need it:`) of PR.
Hope that gives the context to the change I made...

File: cmd/kubelet/app/server_others.go [hunk 2]
```diff
@@ -28,7 +28,25 @@ func checkPermissions() error {
...
+		return err
+	}
+
+	return nil
+}
+
+// checkInitialUserNamespace checks if kubelet is running in the initial user namespace.
+// http://man7.org/linux/man-pages/man7/user_namespaces.7.html
+func checkInitialUserNamespace() error {
+	uidMap, err := ioutil.ReadFile("/proc/self/uid_map")
```
[the remark this answers is not in the document]
Comment (chansuke) [reply cmd/kubelet/app/server_others.go]: @SataQiu I appreciate your review. I have updated the `ioutil.ReadFile`.

File: cmd/kubelet/app/server_others.go [hunk 3]
```diff
@@ -28,7 +28,33 @@ func checkPermissions() error {
...
+// 
+// The initial user namespace is the namespace from which all other namespaces are derived.
+// Running in the initial user namespace ensures that
+// the process has the same view of users and groups as the host system.
+// The function reads the /proc/self/uid_map file, which describes the UID map for the current process.
+// The initial user namespace has a uid_map of "0 0 4294967295".
+func checkInitialUserNamespace() error {
+	uidMap, err := os.ReadFile("/proc/self/uid_map")
+	if err != nil {
+		return err
```
Comment (bart0sh) [inline cmd/kubelet/app/server_others.go:50]: Can you wrap this error?

**Comment by bart0sh** [comment]
@chansuke
> This is vital for maintaining the security and integrity of the system, as it prevents processes running within the Kubelet from potentially gaining unauthorized access to system resources by manipulating their perceived user or group identities.

Can you give an example of this privilege escalation, considering that processes running in the initial namespace have highest priority?

**Comment by chansuke** [comment]
@bart0sh A possible example could be a process that is compromised and attempts to perform unauthorized operations.
Even though the process might be running as a non-root user in the global namespace, within its own user namespace it can remap its UID to 0, effectively giving itself root-like privileges for operations inside its namespace.
These privileges, however, do not extend to the rest of the system. 
By running `Kubelet` in the initial user namespace, we ensure that all processes have consistent user and group ID mappings, which helps prevent such scenarios from occurring.

**Pull request closed without merging**